wordpress-mobile/WordPress-Android
# [Jetpack Focus] Implement reader saved posts ContentProvider and Resolver

Fixes #17289

To test:

1 - Install WordPress and log in; in the Reader save some posts for later and check they are listed in the SAVED tab
2 - Set the flag isFeatureFlagEnabled to true on SharedLoginResolver, UserFlagsResolver and ReaderSavedPostsResolver (this cannot be done using the Debug settings screen because that is only available when the user is already logged in);
2 - Install Jetpack using the same variant (e.g. if installed WordPress version was jalapenoDebug, install Jetpack jalapenoDebug);
3 - Jetpack should login automatically after launching, and home screen should be shown;
4 - Check the expected reader saved posts are in Jetpack under SAVED tab.
5 - Check the above is working with empty saved posts list scenario
6 - Check the track events are fired.

## Regression Notes
1. Potential unintended areas of impact
    None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
    Manual testing and unit tests

3. What automated tests I added (or what prevented me from doing so)
    Adding unit test in a dedicated PR

PR submission checklist:
